### ISSUE TITLE
Automatic Format Conversion happening, but wrong format used in next workflow step

### ISSUE THREAD

**Issue by hepcat72** [galaxyproject/galaxy#8801]
I used the CLIP-Seq workflow from this tutorial:

https://galaxyproject.github.io/training-material/topics/transcriptomics/tutorials/clipseq/tutorial.html

At one point, a step produces a bedgraph, but the next step takes a bigwig and fails with an error indicative that it received a bedgraph. The history however shows that a bigwig was automatically created and if you download and re-upload that file, the next step will succeed if you use it. Also, if I add an explicit conversion step in the workflow, it works.

So it seems there’s a problem with automatic format conversion.

**Comment by mvdbeek** [comment]
There is no step that uses bigwig inputs in the workflow or the tutorial. Could you tell me what you're doing for this failure to appear ?

**Comment by hepcat72** [comment]
Oh geez. I apologize. I linked the wrong tutorial/workflow.  I was working on both a CLIP-Seq and Methylation Seq project at the same time.  It's the methylation-seq tutorial/workflow that had this issue:

https://galaxyproject.github.io/training-material/topics/epigenetics/tutorials/methylation-seq/tutorial.html

There is a text reformatting step on a bedgraph file that is connected to a computeMatrix step that takes bigwig.

**Comment by hepcat72** [comment]
Here are some details...

In the json of the workflow in the tutorial, step 20 (computeMatrix) takes the output of step 19 (a bedgraph file from the "awk/text reformatting" tool) as bigwig input.  In this screenshot, you can see that it gets automatically converted to bigwig:

<img width="274" alt="bedgraphnotbigwig" src="https://user-images.githubusercontent.com/2300532/67430119-9a667a80-f5af-11e9-9b94-e16cd2b49418.png">

Yet, computeMatrix fails with the error:

```
Fatal error: Exit code 1 ()
[bwHdrRead] There was an error while reading in the header!
Traceback (most recent call last):
  File "/galaxy/galaxy-app/tool-dependencies/_conda/envs/__deeptools@2.5.1/bin/computeMatrix", line 11, in <module>
    main(args)
  File "/galaxy/galaxy-app/tool-dependencies/_conda/envs/__deeptools@2.5.1/lib/python3.6/site-packages/deeptools/computeMatrix.py", line 422, in main
    hm.computeMatrix(scores_file_list, args.regionsFileName, parameters, blackListFileName=args.blackListFileName, verbose=args.verbose, allArgs=args)
  File "/galaxy/galaxy-app/tool-dependencies/_conda/envs/__deeptools@2.5.1/lib/python3.6/site-packages/deeptools/heatmapper.py", line 235, in computeMatrix
    chromSizes, _ = getScorePerBigWigBin.getChromSizes(score_file_list)
  File "/galaxy/galaxy-app/tool-dependencies/_conda/envs/__deeptools@2.5.1/lib/python3.6/site-packages/deeptools/getScorePerBigWigBin.py", line 161, in getChromSizes
    fh = pyBigWig.open(fname)
RuntimeError: Received an error during file opening!
```

When I googled that error, I discovered that it can be the result of computeMatrix being supplied a bedgraph - though I don't know this is the case in this instance.  My workaround however was to insert an explicit bedgraph to bigwig conversion in the workflow, which yields a successful outcome:

<img width="275" alt="bedgraphtobigwigworks" src="https://user-images.githubusercontent.com/2300532/67430297-f7fac700-f5af-11e9-9e51-7a0c6bde144a.png">

which suggests that maybe the automatic conversion passes the wrong file to computeMatrix.

Also note that when I imported the workflow from the tutorial, I had the following warnings:

<img width="641" alt="methyl" src="https://user-images.githubusercontent.com/2300532/67430365-11037800-f5b0-11e9-86fd-93fbca35ac6a.png">

And we're running galaxy version 19.01.

**Comment by bernt-matthias** [comment]
Is the input in a collection?

**Comment by hepcat72** [comment]
Yes.

**Comment by bernt-matthias** [comment]
I have a similar issue (automatic unzipping of fastqsanger.gz for Trinity which needs unzipped fastq files). If the FASTQ.gz files are in a collection then the automatic conversion jobs are started, but  trinity still gets the zipped files.

**Comment by mvdbeek** [comment]
Should be fixed by #8932

**Comment by bernt-matthias** [comment]
@hepcat72 please try the fix provided by @mvdbeek and reopen if necessary
